src/code/z_play.c

```diff
@@ -1,81 +1,1545 @@
+
+            if (this->roomCtx.curRoom.enablePosLights || (MREG(93) != 0)) {
+                sp268->enablePosLights = true;
+            }
+
+            Lights_BindAll(sp268, this->lightCtx.listHead, NULL, this);
+            Lights_Draw(sp268, gfxCtx);
+
+            Scene_Draw(this);
+
+            if (this->roomCtx.unk78) {
+                //! FAKE: Regalloc around here. s64 solves the regalloc,
+                //        but much too large on the stack. Need:
+                //        `if (stuff that evaluates to 0 but not trivially)`
+                s64 roomDrawFlags = 3;
+
+                Room_Draw(this, &this->roomCtx.curRoom, roomDrawFlags);
+                Room_Draw(this, &this->roomCtx.prevRoom, roomDrawFlags);
+            }
+
+            if (this->skyboxCtx.skyboxShouldDraw) {
+                Vec3f sp78;
+
+                if (1) {}
+                Camera_GetQuakeOffset(&sp78, GET_ACTIVE_CAM(this));
+                SkyboxDraw_Draw(&this->skyboxCtx, gfxCtx, this->skyboxId, 0, this->view.eye.x + sp78.x,
+                                this->view.eye.y + sp78.y, this->view.eye.z + sp78.z);
+            }
+
+            // envCtx.precipitation[PRECIP_RAIN_CUR]
+            if (this->envCtx.unk_F2[1] != 0) {
+                Environment_DrawRain(this, &this->view, gfxCtx);
+            }
+        }
+
+        Environment_FillScreen(gfxCtx, 0, 0, 0, this->bgCoverAlpha, 1);
+
+        Actor_DrawAll(this, &this->actorCtx);
+
+        if (!this->envCtx.sunMoonDisabled) {
+            sp25C.x = this->view.eye.x + this->envCtx.unk_4;
+            sp25C.y = this->view.eye.y + this->envCtx.unk_8;
+            sp25C.z = this->view.eye.z + this->envCtx.unk_C;
+            Environment_DrawSunLensFlare(this, &this->envCtx, &this->view, gfxCtx, sp25C);
+        }
+
+        Environment_DrawCustomLensFlare(this);
+
+        if (R_PLAY_FILL_SCREEN_ON) {
+            Environment_FillScreen(gfxCtx, R_PLAY_FILL_SCREEN_R, R_PLAY_FILL_SCREEN_G, R_PLAY_FILL_SCREEN_B,
+                                   R_PLAY_FILL_SCREEN_ALPHA, 3);
+        }
+
+        switch (this->envCtx.fillScreen) {
+            case 1:
+                Environment_FillScreen(gfxCtx, this->envCtx.screenFillColor[0], this->envCtx.screenFillColor[1],
+                                       this->envCtx.screenFillColor[2], this->envCtx.screenFillColor[3], 3);
+                break;
+            default:
+                break;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80167DE4.s")
+        if (this->envCtx.sandstormState != 0) {
+            Environment_DrawSandstorm(this, this->envCtx.sandstormState);
+        }
+
+        if (this->unk_18876 != 0) {
+            Environment_FillScreen(gfxCtx, 0, 0, 0, this->unk_18876, 3);
+        }
+
+        DebugDisplay_DrawObjects(this);
+        Play_DrawMotionBlur(this);
+
+        if (((R_PAUSE_MENU_MODE == 1) || (gTrnsnUnkState == 1)) || (R_PICTOGRAPH_PHOTO_STATE == 1)) {
```

Comment:
```suggestion
        if (((R_PAUSE_MENU_MODE == 1) || (gTrnsnUnkState == 1)) || (R_PICTOGRAPH_PHOTO_STATE == PICTOGRAPH_PHOTO_STATE_START)) {
```
There's a few of these missed for `R_PICTOGRAPH_PHOTO_STATE`, could you double check these?